README.md
# HmtVar  

[HmtVar](https://www.hmtvar.uniba.it) is a database hosting data about human mitochondrial variants, integrated with their pathogenicity predictions. More details can be found on the official paper, whose DOI is [10.1093/nar/gky1024](https://doi.org/10.1093/nar/gky1024).  

## Installation  

1. Create a new virtual environment (Python 3): `virtualenv -p python3.6 venv`  
2. Activate the new virtual environment: `source venv/bin/activate`  
3. Install all the required packages: `pip install -r requirements.txt`  
3a. Will need to have the `update/data/tables/` folder available, with all the required tables in it.  
4. Export the needed variables: `export FLASK_APP=app:app`  
5. Create the db: `flask create-db`  
6. Update the db: `flask update-db`  
7. Migrate the db: `flask migrate-db`  
8. Update the additional files used by [HmtNote](https://github.com/robertopreste/HmtNote): `flask update-hmtnote`  

Everytime you need to update the database with new data (from the tables in `update/data/tables/`), you need to repeat steps 4, 6, 7, 8 (**not 5!**).

## Launch HmtVar instance  

After that, you are ready to launch the HmtVar database using `python run.py`.  

The database will then be available at `127.0.0.1:5000`.  

Currently, HmtVar is deployed using gunicorn; if you need to modify this, you should already know how to do it.  

## Updates  

Details about each update of the public HmtVar instance can be found in the [CHANGELOG](/CHANGELOG.md) file.  
Details about the data updating protocol can be found in the [UPDATE.md](/update/UPDATE.md) file.  

## Citing HmtVar  

HmtVar can be cited as:  

>Preste R, Vitale O, Clima R, Gasparre G, Attimonelli M  
>
>**HmtVar: a new resource for human mitochondrial variations and pathogenicity data.**  
>
>[*Nucleic Acids Res. 2018 Oct 29.*](https://doi.org/10.1093/nar/gky1024)  




CHANGELOG.md
# HmtVar - CHANGELOG  
---  

# July 31 2019  

* Update variability and allele frequency from latest HmtDB update (July 2019)
* Update pathogenicity accordingly   

## Statistics  

### Variants  

Total variants: **40976**  

#### Locus type  

| Locus Type | Variants | 
| ---------- | -------- | 
| CDS        | 34297    | 
| tRNA       | 4547     | 
| Regulatory | 1329     | 
| rRNA       | 803      | 

#### Pathogenicity  

| Pathogenicity tier | Variants | 
| ------------------ | -------- | 
| Pathogenic         | 18899    | 
| Likely polymorphic | 9666     | 
| Polymorphic        | 152      | 
| Likely pathogenic  | 36       | 
| VUS                | 15       | 

### Genomes  

* Complete genomes: 49304  
	* Healthy: 44058  
	* Patients: 5246  

___  


# July 26 2019  

* Add 53 new variants
* Update disease_score and pathogenicity for CDS variants  

## Statistics  

### Variants  

Total variants: **40976**  

#### Locus type  

| Locus Type | Variants | 
| ---------- | -------- | 
| CDS        | 34297    | 
| tRNA       | 4547     | 
| Regulatory | 1329     | 
| rRNA       | 803      | 

#### Pathogenicity  

| Pathogenicity tier | Variants | 
| ------------------ | -------- | 
| Pathogenic         | 18898    | 
| Likely polymorphic | 9660     | 
| Polymorphic        | 162      | 
| Likely pathogenic  | 38       | 
| VUS                | 19       | 

### Genomes  

* Complete genomes: 46766  
	* Healthy: 42057  
	* Patients: 4709  

___  

## May 2019 

* Update aa_change field in Main table to fix variants with wrong aa change annotation  

### Statistics  

#### Variants  

Total variants: **40923**  

##### Locus type  

| Locus Type | Variants | 
| ---------- | -------- | 
| CDS        | 34297    | 
| tRNA       | 4547     | 
| Regulatory | 1291     | 
| rRNA       | 788      | 

##### Pathogenicity  

| Pathogenicity tier | Variants | 
| ------------------ | -------- | 
| Pathogenic         | 18857    | 
| Likely polymorphic | 9708     | 
| Polymorphic        | 154      | 
| Likely pathogenic  | 34       | 
| VUS                | 15       | 

#### Genomes  

* Complete genomes: 46766  
	* Healthy: 42057  
	* Patients: 4709  

___  

## February 2019  

* Update haplogroups field in Main table from the latest MToolBox `haplogroups.txt` file    
* Update Predict and CrossRef tables from new Clinvar data  

### Statistics  

#### Variants  

Total variants: **40923**  

##### Locus type  

| Locus Type | Variants | 
| ---------- | -------- | 
| CDS        | 34297    | 
| tRNA       | 4547     | 
| Regulatory | 1291     | 
| rRNA       | 788      | 

##### Pathogenicity  

| Pathogenicity tier | Variants | 
| ------------------ | -------- | 
| Pathogenic         | 18857    | 
| Likely polymorphic | 9708     | 
| Polymorphic        | 154      | 
| Likely pathogenic  | 34       | 
| VUS                | 15       | 

#### Genomes  

* Complete genomes: 46766  
	* Healthy: 42057  
	* Patients: 4709  

___  

## January 2019  

* Calculate new variability and allele frequencies from HmtDB October 2018 update  
* Update pathogenicity tiers based on new allele frequencies  
* Update Plasmy and CrossRef tables from new Mitomap data  

### Statistics  

#### Variants  

Total variants: **40923**  

##### Locus type  

| Locus Type | Variants | 
| ---------- | -------- | 
| CDS        | 34297    | 
| tRNA       | 4547     | 
| Regulatory | 1291     | 
| rRNA       | 788      | 

##### Pathogenicity  

| Pathogenicity tier | Variants | 
| ------------------ | -------- | 
| Pathogenic         | 18857    | 
| Likely polymorphic | 9708     | 
| Polymorphic        | 154      | 
| Likely pathogenic  | 34       | 
| VUS                | 15       | 

#### Genomes  

* Complete genomes: 46766  
	* Healthy: 42057  
	* Patients: 4709  


app/static/dbdata.py
cds_loci = [('MT-ATP6', 'MT-ATP6'), ('MT-ATP8', 'MT-ATP8'), ('MT-CO1', 'MT-CO1'), ('MT-CO2', 'MT-CO2'), ('MT-CO3', 'MT-CO3'), ('MT-CYB', 'MT-CYB'), ('MT-ND1', 'MT-ND1'), ('MT-ND2', 'MT-ND2'), ('MT-ND3', 'MT-ND3'), ('MT-ND4', 'MT-ND4'), ('MT-ND4L', 'MT-ND4L'), ('MT-ND5', 'MT-ND5'), ('MT-ND6', 'MT-ND6')]
reg_loci = [('MT-DLOOP', 'MT-DLOOP'), ('MT-NC10', 'MT-NC10'), ('MT-NC2', 'MT-NC2'), ('MT-NC3', 'MT-NC3'), ('MT-NC5', 'MT-NC5'), ('MT-NC7', 'MT-NC7'), ('MT-NC9', 'MT-NC9'), ('MT-ORIL', 'MT-ORIL')]
rrna_loci = [('MT-RNR1', 'MT-RNR1'), ('MT-RNR2', 'MT-RNR2')]
trna_loci = [('MT-TA', 'MT-TA'), ('MT-TC', 'MT-TC'), ('MT-TD', 'MT-TD'), ('MT-TE', 'MT-TE'), ('MT-TF', 'MT-TF'), ('MT-TG', 'MT-TG'), ('MT-TH', 'MT-TH'), ('MT-TI', 'MT-TI'), ('MT-TK', 'MT-TK'), ('MT-TL1', 'MT-TL1'), ('MT-TL2', 'MT-TL2'), ('MT-TM', 'MT-TM'), ('MT-TN', 'MT-TN'), ('MT-TP', 'MT-TP'), ('MT-TQ', 'MT-TQ'), ('MT-TR', 'MT-TR'), ('MT-TS1', 'MT-TS1'), ('MT-TS2', 'MT-TS2'), ('MT-TT', 'MT-TT'), ('MT-TV', 'MT-TV'), ('MT-TW', 'MT-TW'), ('MT-TY', 'MT-TY')]
all_loci = [('MT-ATP6', 'MT-ATP6'), ('MT-ATP8', 'MT-ATP8'), ('MT-CO1', 'MT-CO1'), ('MT-CO2', 'MT-CO2'), ('MT-CO3', 'MT-CO3'), ('MT-CYB', 'MT-CYB'), ('MT-DLOOP', 'MT-DLOOP'), ('MT-NC10', 'MT-NC10'), ('MT-NC2', 'MT-NC2'), ('MT-NC3', 'MT-NC3'), ('MT-NC5', 'MT-NC5'), ('MT-NC7', 'MT-NC7'), ('MT-NC9', 'MT-NC9'), ('MT-ND1', 'MT-ND1'), ('MT-ND2', 'MT-ND2'), ('MT-ND3', 'MT-ND3'), ('MT-ND4', 'MT-ND4'), ('MT-ND4L', 'MT-ND4L'), ('MT-ND5', 'MT-ND5'), ('MT-ND6', 'MT-ND6'), ('MT-ORIL', 'MT-ORIL'), ('MT-RNR1', 'MT-RNR1'), ('MT-RNR2', 'MT-RNR2'), ('MT-TA', 'MT-TA'), ('MT-TC', 'MT-TC'), ('MT-TD', 'MT-TD'), ('MT-TE', 'MT-TE'), ('MT-TF', 'MT-TF'), ('MT-TG', 'MT-TG'), ('MT-TH', 'MT-TH'), ('MT-TI', 'MT-TI'), ('MT-TK', 'MT-TK'), ('MT-TL1', 'MT-TL1'), ('MT-TL2', 'MT-TL2'), ('MT-TM', 'MT-TM'), ('MT-TN', 'MT-TN'), ('MT-TP', 'MT-TP'), ('MT-TQ', 'MT-TQ'), ('MT-TR', 'MT-TR'), ('MT-TS1', 'MT-TS1'), ('MT-TS2', 'MT-TS2'), ('MT-TT', 'MT-TT'), ('MT-TV', 'MT-TV'), ('MT-TW', 'MT-TW'), ('MT-TY', 'MT-TY')]
cds_vars = 34297
reg_vars = 1291
rrna_vars = 788
trna_vars = 4547
p_likely_pathogenic = 34
p_likely_polymorphic = 9708
p_pathogenic = 18857
p_polymorphic = 154
p_vus = 15
cds_patho = [26, 18691, 126, 5387]
reg_patho = [0, 0, 0, 0]
rrna_patho = [0, 0, 0, 0]
trna_patho = [8, 166, 28, 4321, 15]
loci_lengths = {'MT-ATP6': 681, 'MT-ATP8': 207, 'MT-CO1': 1542, 'MT-CO2': 684, 'MT-CO3': 784, 'MT-CYB': 1141, 'MT-DLOOP1': 576, 'MT-DLOOP2': 546, 'MT-ND1': 956, 'MT-ND2': 1042, 'MT-ND3': 346, 'MT-ND4': 1378, 'MT-ND4L': 297, 'MT-ND5': 1812, 'MT-ND6': 525, 'MT-NC1': 2, 'MT-NC2': 1, 'MT-NC3': 10, 'MT-NC4': 1, 'MT-NC5': 12, 'MT-NC6': 1, 'MT-NC7': 25, 'MT-NC8': 1, 'MT-NC9': 4, 'MT-NC10': 1, 'MT-ORIL': 78, 'MT-RNR1': 954, 'MT-RNR2': 1559, 'MT-TL1': 75, 'MT-TA': 69, 'MT-TR': 65, 'MT-TN': 73, 'MT-TD': 68, 'MT-TC': 66, 'MT-TQ': 72, 'MT-TE': 69, 'MT-TG': 68, 'MT-TH': 69, 'MT-TI': 69, 'MT-TL2': 71, 'MT-TK': 70, 'MT-TM': 68, 'MT-TF': 71, 'MT-TP': 69, 'MT-TS2': 59, 'MT-TS1': 72, 'MT-TT': 66, 'MT-TW': 68, 'MT-TY': 66, 'MT-TV': 69}
single_locus_vars = {'MT-ATP6': 2049, 'MT-ATP8': 621, 'MT-CO1': 4630, 'MT-CO2': 2054, 'MT-CO3': 2361, 'MT-CYB': 3491, 'MT-DLOOP1': 1133, 'MT-DLOOP2': 1133, 'MT-ND1': 2870, 'MT-ND2': 3132, 'MT-ND3': 1038, 'MT-ND4': 4140, 'MT-ND4L': 891, 'MT-ND5': 5443, 'MT-ND6': 1577, 'MT-NC1': 0, 'MT-NC2': 1, 'MT-NC3': 4, 'MT-NC4': 0, 'MT-NC5': 3, 'MT-NC6': 0, 'MT-NC7': 6, 'MT-NC8': 0, 'MT-NC9': 1, 'MT-NC10': 1, 'MT-ORIL': 142, 'MT-RNR1': 313, 'MT-RNR2': 475, 'MT-TL1': 228, 'MT-TA': 207, 'MT-TR': 195, 'MT-TN': 219, 'MT-TD': 204, 'MT-TC': 198, 'MT-TQ': 220, 'MT-TE': 207, 'MT-TG': 204, 'MT-TH': 207, 'MT-TI': 211, 'MT-TL2': 213, 'MT-TK': 210, 'MT-TM': 204, 'MT-TF': 213, 'MT-TP': 205, 'MT-TS2': 179, 'MT-TS1': 211, 'MT-TT': 199, 'MT-TW': 206, 'MT-TY': 200, 'MT-TV': 207}
latest_update = 'June 2019'
